### PR TITLE
Adding dll support for win32

### PR DESCRIPTION
**Pull request by rakhad**

### CODE REVIEW

File: libHttpClient.props [hunk 1]
```diff
@@ -34,13 +34,31 @@
...
 
   <PropertyGroup>
     <HCBuildRoot Condition="'$(HCBuildRoot)'==''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), build.root))\</HCBuildRoot>
     <libHttpClientInclude>$(HCBuildRoot)Include\</libHttpClientInclude>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(useDll)'=='true'">
+    <importLibPath></importLibPath>
+    <dllPath></dllPath>
+
```
Comment (jasonsandlin) [inline libHttpClient.props]: remove

File: libHttpClient.props [hunk 2]
```diff
@@ -34,13 +34,31 @@
...
   <PropertyGroup>
     <HCBuildRoot Condition="'$(HCBuildRoot)'==''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), build.root))\</HCBuildRoot>
     <libHttpClientInclude>$(HCBuildRoot)Include\</libHttpClientInclude>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(useDll)'=='true'">
+    <importLibPath></importLibPath>
+    <dllPath></dllPath>
+    
+    <importLibPath Condition="'$(HCLibPlatformType)'=='Win32'">$(HCBuildRoot)\Binaries\$(Configuration)\$(Platform)\libHttpClient.Win32.C\libHttpClient.Win32.C.lib</importLibPath>
```
Comment (jasonsandlin) [inline libHttpClient.props]: >importLibPath [](start = 5, length = 13)

pascal case -> ImportLibPath and DllPath

File: libHttpClient.props [hunk 3]
```diff
@@ -34,13 +34,31 @@
...
   </PropertyGroup>
 
   <PropertyGroup>
     <HCBuildRoot Condition="'$(HCBuildRoot)'==''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), build.root))\</HCBuildRoot>
     <libHttpClientInclude>$(HCBuildRoot)Include\</libHttpClientInclude>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(useDll)'=='true'">
+    <importLibPath></importLibPath>
+    <dllPath></dllPath>
```
Comment (jicailiu) [inline libHttpClient.props]: no need

File: libHttpClient.props [hunk 4]
```diff
@@ -34,13 +34,31 @@
     <HCLibToolset Condition="'$(HCLibPlatformType)'=='Android'">141</HCLibToolset>
 
     <HCProjectName>libHttpClient.$(HCLibToolset).$(HCLibPlatformType).C</HCProjectName>
+    <HCProjectName Condition="'$(HCLibPlatformType)'=='Win32'">libHttpClient.$(HCLibPlatformType).C</HCProjectName>
+
+    <useDll>false</useDll>
```
Comment (jplafonta) [inline libHttpClient.props]: >useDll>false</useDll> [](start = 5, length = 21)

Actually no need for this either

File: Build/libHttpClient.Win32.C/libHttpClient.Win32.C.vcxproj [hunk 5]
```diff
@@ -73,9 +73,11 @@
       <GenerateDebugInformation>DebugFull</GenerateDebugInformation>
       <GenerateWindowsMetadata>false</GenerateWindowsMetadata>
       <FullProgramDatabaseFile Condition="'$(Configuration)'=='Debug'">true</FullProgramDatabaseFile>
+      <ModuleDefinitionFile>libHttpClient.Win32.C.def</ModuleDefinitionFile>
     </Link>
     <Lib>
```
Comment (lubeltra) [inline Build/libHttpClient.Win32.C/libHttpClient.Win32.C.vcxproj]: Is this still relevant if we are building as a dll?

Comment (rakhad) [reply Build/libHttpClient.Win32.C/libHttpClient.Win32.C.vcxproj]: I can test removing it

---
In reply to: [225740181](https://github.com/Microsoft/libHttpClient/pull/308#discussion_r225740181) [](ancestors = 225740181)

Comment (rakhad) [reply Build/libHttpClient.Win32.C/libHttpClient.Win32.C.vcxproj]: Tested using the samples and it works without the <lib> tag

---
In reply to: [225740579](https://github.com/Microsoft/libHttpClient/pull/308#discussion_r225740579) [](ancestors = 225740579,225740181)

File: libHttpClient.props [hunk 6]
```diff
@@ -34,13 +34,31 @@
...
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(useDll)'=='true'">
+    <importLibPath></importLibPath>
+    <dllPath></dllPath>
+    
+    <importLibPath Condition="'$(HCLibPlatformType)'=='Win32'">$(HCBuildRoot)\Binaries\$(Configuration)\$(Platform)\libHttpClient.Win32.C\libHttpClient.Win32.C.lib</importLibPath>
+    <dllPath Condition="'$(HCLibPlatformType)'=='Win32'">$(HCBuildRoot)\Binaries\$(Configuration)\$(Platform)\libHttpClient.Win32.C\libHttpClient.Win32.C.dll</dllPath>
+      
+    <AdditionalDependencies>%(importLibPath);$(AdditionalDependencies);</AdditionalDependencies>
```
Comment (rakhad) [inline libHttpClient.props]: ><AdditionalDependencies>%(importLibPath);$(AdditionalDependencies);</AdditionalDependencies> [](start = 4, length = 92)

Break ImportLibPath into additional lib paths and additional dependencies

**Comment by rakhad** [comment]
> <?xml version="1.0" encoding="utf-8"?>

Make it work without source

---
Refers to: libHttpClient.props:1 in 821ee9e. [](commit_id = 821ee9e19457f9b87278100a68f679dd149c7738, deletion_comment = False)

**Pull request merged**